DAQ_System/Source/main.cpp

```diff
@@ -1,23 +1,140 @@
+            sample size of n when stationary. this only needs to be done once so this if statement wont really every repeat.
+             */
+            if(offsetStatus != true)
+            {
+                averageOffsetCalc(offset, sum);
+                offsetStatus = true; // offset is set so this setting makes it so it wont callibrate every single time.
+            }
+                
+
+            degreesPerSecondConversion(degreesPerSecond, offset, gyroData);
+            radiansPerSecondConversion(radiansPerSecond, degreesPerSecond);
+
+            /* this is the ANSI escape sequences 
+            \033 is the escape character, which signals the start of an escape sequence. 
+            [2J is the clear screen command, which clears the entire screen.
+            [1;1H is the cursor position command, which moves the cursor to a specified     
+            */
+                   
+            //raw data from the gyroscope
+            //printf("\033[2J\033[1;1H"); 
+            // printf("tx: %d \n",gyroData[0] ); 
+            //printf("ty: %d \n",gyroData[1] );
+            //printf("tz: %d \n",gyroData[2] );
+            // displays the average offset used for callibration of the gyroscope
+            //printf("the average of the X-axis is: %d \n", averageOffset[0]);
+            //printf("the average of the y-axis is: %d \n", averageOffset[1]);
+            //printf("the average of the z-axis is: %d \n", averageOffset[2]);
+            // displays the callibrated data of the gyroscope in dps
+            //printf("callibration of X(in dps): %d \n",degreesPerSecond[0]);
+            //printf("callibration of Y(in dps): %d \n",degreesPerSecond[1]);
+            //printf("callibration of Z(in dps): %d \n",degreesPerSecond[2]);
+            // displays the callibrated data of the gyroscope in radians per second
+            //printf("callibration of x(in rad/s) %.5f \n",radiansPerSecond[0]);
+            //printf("callibration of y(in rad/s) %.5f \n",radiansPerSecond[1]);
+            //printf("callibration of z(in rad/s) %.5f \n",radiansPersecond[2]);
+            
+  
+                    
+
+            //thread_sleep_for(500);
+
+                
+
+        } 
+    }     
+            
+}
+
+
+
+void averageOffsetCalc(short Offset[], short sum[])
+{
+  // collecting the sampled data for the offset average
+  short gyroData[3] = {0};
+  int averageSampleSize = 15;
+
+    for (int i = 0; i <= averageSampleSize; i++) 
+    {
+        gyro.read(gyroData);
+        sum[0] += gyroData[0];
+        sum[1] += gyroData[1];
+        sum[2] += gyroData[2];
+        // the if statment is used to find the average and will be calculated when all sample data is retrieved
+        if( i == averageSampleSize )
+        {
+            Offset[0] = sum[0]/averageSampleSize;
+            Offset[1] = sum[1]/averageSampleSize;
+            Offset[2] = sum[2]/averageSampleSize;
+        }
     }
```

Comment:
Is it necessary for the `if` statement to be inside of the `for` loop? The loop will end after `i` is equal to `averageSampleSize`, which is what the `if` condition is checking for

Reply:
lol, you're right. I was probably half asleep when writing that or something. I will remove the if statement entirely and bring the offset-averaging equation out of the loop